src/lib/QueryType/SubtreeQueryType.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformAdminUi\QueryType;
+
+use eZ\Publish\API\Repository\PermissionResolver;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\QueryType\OptionsResolverBasedQueryType;
+use eZ\Publish\Core\QueryType\QueryType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+abstract class SubtreeQueryType extends OptionsResolverBasedQueryType implements QueryType
+{
+    protected const OWNED_OPTION_NAME = 'owned';
+    protected const SUBTREE_OPTION_NAME = 'subtree';
+
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
+    protected $configResolver;
+
+    /** @var \eZ\Publish\API\Repository\PermissionResolver */
+    private $permissionResolver;
+
+    public function __construct(
+        ConfigResolverInterface $configResolver,
+        PermissionResolver $permissionResolver
+    ) {
+        $this->configResolver = $configResolver;
+        $this->permissionResolver = $permissionResolver;
+    }
+
+    public function doGetQuery(array $parameters): Query
+    {
+        $contentSubtreePath = $parameters[SubtreeQueryType::SUBTREE_OPTION_NAME] ?? $this->getSubtreePathFromConfiguration();
```

Comment:
```php
$parameters[SubtreeQueryType::SUBTREE_OPTION_NAME] ?? $this->getSubtreePathFromConfiguration();
```

This should be part of the options resolver configuration (`$this->getSubtreePathFromConfiguration()` is the default value for `subtree` option).